databases/week2/ErDiagram.sql

```diff
@@ -0,0 +1,91 @@
+
+
+SET NAMES utf8mb4;
+
+CREATE TABLE `User` (
+  `id` INT NOT NULL AUTO_INCREMENT,
+  `email` VARCHAR(255) NOT NULL,
+  `password` VARCHAR(32) NOT NULL,
+  `create_time` DATETIME NOT NULL,
+  `name` VARCHAR(45) NOT NULL,
+  PRIMARY KEY (`id`))
+ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+CREATE TABLE `Occassion` (
+  `id` INT NOT NULL AUTO_INCREMENT,
+  `name` VARCHAR(255) NOT NULL,
+  PRIMARY KEY (`id`))
+ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+
+CREATE TABLE `Payment` (
+  `id` INT NOT NULL AUTO_INCREMENT,
+  `amount` FLOAT NOT NULL,
+  `user_id` INT NOT NULL,
+  `Order_id` INT NOT NULL,
+  PRIMARY KEY (`id`),
+  CONSTRAINT `fk_user`
+    FOREIGN KEY (`user_id`)
+    REFERENCES `User` (`id`)
+    ON DELETE NO ACTION
+    ON UPDATE NO ACTION,
+  CONSTRAINT `fk_Payment_Order`
+    FOREIGN KEY (`Order_id`)
+    REFERENCES `Order` (`id`)
+    ON DELETE CASCADE
+    ON UPDATE CASCADE)
+ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+
+CREATE TABLE `Order` (
+  `id` INT NOT NULL AUTO_INCREMENT,
+  `user_id` INT NOT NULL,
+  `create_date` DATETIME NOT NULL,
+  `status` VARCHAR(25) NOT NULL,
+  `Payment_id` INT NOT NULL,
+  PRIMARY KEY (`id`),
+  CONSTRAINT `fk_user`
+    FOREIGN KEY (`user_id`)
+    REFERENCES `User` (`id`)
+    ON DELETE NO ACTION
+    ON UPDATE NO ACTION,
+  CONSTRAINT `fk_Order_Payment`
+    FOREIGN KEY (`Payment_id`)
+    REFERENCES `Payment` (`id`)
+    ON DELETE CASCADE
+    ON UPDATE CASCADE)
+ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+
+CREATE TABLE `Cake` (
+  `id` INT NOT NULL AUTO_INCREMENT,
+  `theme` VARCHAR(45) NOT NULL,
+  `flavor` VARCHAR(45) NOT NULL,
+  `frosting_type` VARCHAR(45) NOT NULL,
+  `type` VARCHAR(45) NOT NULL,
+  `Order_id` INT NOT NULL,
+  PRIMARY KEY (`id`),
+   CONSTRAINT `fk_Cake_Order`
+    FOREIGN KEY (`Order_id`)
+    REFERENCES `Order` (`id`)
+    ON DELETE CASCADE
+    ON UPDATE CASCADE)
+ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;a
+
+
+CREATE TABLE `Cake_Occassion` (
+  `Cake_id` INT NOT NULL,
+  `Occassion_id` INT NOT NULL,
+  PRIMARY KEY (`Cake_id`, `Occassion_id`),
+     CONSTRAINT `fk_Cake_Occassion_idx1`
+    FOREIGN KEY (`Cake_id`)
+    REFERENsCES `Cake` (`id`)
+    ON DELETE CASCADE
+    ON UPDATE CASCADE,
+  CONSTRAINT `fk_Cake_Occassion_idx2`
+    FOREIGN KEY (`Occassion_id`)
+    REFERENCES `Occassion` (`id`)
+    ON DELETE CASCADE
+    ON UPDATE CASCADE)
+ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+}
```

Comment:
this last line gives an error.

Overall, the file to create the tables is ok, but there are a few mistakes, could you fix them and make sure the file runs please? 
Also, don't forget to create a database and create the tables there :)

Regarding the DB design (ER diagram), i have a few questions:
  1. Can one type of cake only be part of one order?
  2. Can't i get chocolate cake in multiple orders?
  3. What's the relationship between orders and payments?
  4. Why having order_id in the payment table and payment_id in the order table?

Reply:
1. No. The relationship is - one order can contain multiple cakes. Should I also add another relationship that one cake can be part of multiple orders?
2. Could you elaborate a little, now I am thinking that maybe there should be a many to many relationship between cake and order.
3.  An order has a payment or a payment belongs to an order
4. as mentioned in point 3 above, they payment and order have a 1:1 relationship

Reply:
Regarding points 1 and 2, if i understand your database correctly, the relationship between cake and order should be many to many indeed. In the sense that one type of cake can be part of many orders and one order can include many cakes. Does it make sense?

Regarding points 3 and 4. In general, when you have a 1:1 relationship you can just merge the tables. But if for some reason you prefer to still have separate tables, one foreign key in one of the tables is enough. Does it make sense?